storage_service/locations/models/space.py

```diff
@@ -795,6 +800,45 @@ def _delete_path_local(self, delete_path):
             LOGGER.warning("Error deleting package %s", delete_path, exc_info=True)
             raise
 
+    @staticmethod
+    def _retrieve_dirs_to_delete(delete_path):
+        """Given an absolute path to an Archivematica AIP (file or
+        uncompressed directory), retrieve the base and relative
+        directories that contain it.
+
+        :param delete_path: absolute path to an Archivematica package
+            (path-string)
+        :return: quad_directory_structure (path-string), base_dir
+            (path-string) (tuple)
+        """
+        UUID_MASK = os.path.join(
+            "1111", "1111", "1111", "1111", "1111", "1111", "1111", "1111"
+        )
+        absolute_path_to_aip = os.path.split(delete_path)[0]
+        base_dir = absolute_path_to_aip[: -len(UUID_MASK)]
+        quad_dir_structure = absolute_path_to_aip.split(base_dir)[1]
+        return quad_dir_structure, base_dir
+
+    def _delete_quad_dir_structure(self, delete_path):
+        """Given an absolute path to a package which is expected to have
+        been deleted, split the object to then enable deletion of the
+        quad-directory structure that the package used to sit within.
+
+        :param delete_path: absolute path to an Archivematica package
+            (path-string)
+        """
+        quad_dir, base_dir = self._retrieve_dirs_to_delete(delete_path)
+        utils.removedirs(quad_dir, base=base_dir)
+
+    def _delete_path_local(self, delete_path):
+        """
+        Deletes `delete_path` in this space, assuming it is locally accessible.
+        """
+        if not os.path.exists(delete_path):
+            return
+        self._del_package(delete_path)
+        self._delete_quad_dir_structure(delete_path)
```

Comment:
I am happy to put a `LOGGER.debug` in here. That only shows up in the most verbose of logs and I cannot imagine this will create a lot of noise. Good suggestion.